Sources/NIOSSH/TransportProtection/SSHTransportProtection.swift

```diff
@@ -87,10 +87,10 @@ protocol NIOSSHTransportProtection: AnyObject {
     /// length, the padding, or the MAC), and update source to indicate the consumed bytes.
     /// It must also perform any integrity checking that
     /// is required and throw if the integrity check fails.
-    func decryptAndVerifyRemainingPacket(_ source: inout ByteBuffer) throws -> ByteBuffer
+    func decryptAndVerifyRemainingPacket(_ source: inout ByteBuffer, sequenceNumber: UInt32) throws -> ByteBuffer
 
     /// Encrypt an entire outbound packet
-    func encryptPacket(_ packet: NIOSSHEncryptablePayload, to outboundBuffer: inout ByteBuffer) throws
+    func encryptPacket(_ packet: NIOSSHEncryptablePayload, sequenceNumber: UInt32, to outboundBuffer: inout ByteBuffer) throws
```

Comment:
One thing I haven't solved yet, is how to use `NIOSSHEncryptablePayload` to write tests for custom transport implementations. It uses internal type (SSHMessage), so there is no way to initialize it. I see three options:
1. Make it a protocol
2. Make it an enum (`SSHMessage|ByteBuffer)
3. Make `encryptPacket` similar to decrypt, where we will be passing an `inout` buffer with message already written there expecting client to encrypt it inplace
wdyt?

Reply:
(right now I'm in favour of 3 tbh)

Reply:
I think many implementation will begin with:
```
        let packetLengthIndex = outboundBuffer.writerIndex
        let packetLengthLength = MemoryLayout<UInt32>.size
        let packetPaddingIndex = outboundBuffer.writerIndex + packetLengthLength
        let packetPaddingLength = MemoryLayout<UInt8>.size

        outboundBuffer.moveWriterIndex(forwardBy: packetLengthLength + packetPaddingLength)
```
maybe we should abstract in serializer?

Reply:
I suspect (3) is the right approach, yes. We can also abstract the write pattern in the serializer as well. Can I suggest that we do that as a separate PR though to avoid this one growing further?

Reply:
agreed 👍